### ISSUE TITLE
Privacy risk: can third-parties distributed and colluding on enough sites still infer top user topics? Can this lead to risk of unicity? I.e. allow tracking individuals

### ISSUE THREAD

**Issue by aberke** [patcg-individual-drafts/topics#13]
Suppose there are P third-parties {p1, p2, p3, …} who each have their code available to call the Topics API on many of the sites users visit. And suppose they either share information, or are even the same higher level party. I.e. ad-tech company A has servers p1,p2,p3,... all calling the Topics API on each of these sites. Each p calling the API on a given site sees a random 3 of the 5 top topics, with a 5% chance of the random topic. With enough simultaneous calls from p1,p2,p3…, they can learn the top 5 topics for that user by what is probabilistically returned.

For each site they might then create  a pseudo-identifier that concatenates the top topics. E.g. If they learn the top topics for the user returned by that site are t1,t2,t3,t4,t5, then they might construct a string "t1-t2-t3-t4-t5".

Assuming these third parties are well distributed across the sites with the various topics a user visits, they might have full access to the user's topics. And they might then be able to gather a consistent pseudo-identifier for a user across the sites they visit ("t1-t2-t3-t4-t5").

Such pseudo-identifiers will likely be shared across many users who share the same interests, in any given week. Yet with 350 topics, there are 350**5 top topic combinations. Could some be unique?
Even if this kind of unicity is rare, the top topics change each week/epoch, and after a sufficient number of weeks, the sequence of pseudo-identifiers collected across these weeks might uniquely identify users. 
i.e. this could allow the kind of cross-site tracking that removing third-party cookies is meant to do away with.

An assumption here is that the separate calls by p1,p2,p3,... for when a user visits a domain can be connected by the caller. This might be done by combining the timing of the calls with fingerprinting data. 

This is a bit convoluted and makes a lot of assumptions. Are any of the assumptions and privacy concerns valid?

**Comment by jkarlin** [comment]
Speaking at a high level, yes, with enough time, it may be probabilistically possible to identify a user across sites using enough a large list of topics. That is why we've tried so hard to reduce the cross-site fingerprintable information per epoch.

In regards to your example, I think there is some confusion. On the same site, up to 3 topics can be learned (the current week's and one from each of the previous two) per week. No matter how many callers there are on the page. They'll all have the same topic returned (assuming they've observed it). In the following week, only one additional topic can be learned, since the previous two weeks were already learned. Ultimately, this is a discussion about the rate at which topics can be acquired, and the rate at which one might reidentify the same user across two different sites, which again, we've tried to keep to a minimum.
